models.py

```diff
@@ -100,17 +101,29 @@ def annotations_by_books_and_prints(self, group_amount=50):
 
         return (books,prints)
 
+    def format_trp_id(self, trp_id):
+        return '%04d' % int(trp_id)
 
     def _get_trp_id(self):
-        last_bio = Biography.objects.order_by('-trp_id')[0]
+        ordered_bios = Biography.objects.order_by('-trp_id')
+        index = 0
+        last_bio = ordered_bios[index]
+        while last_bio.trp_id != self.format_trp_id(last_bio.trp_id):
+            last_bio = ordered_bios[index+1]
         last_trp_id = int(last_bio.trp_id)
         new_trp_id = last_trp_id + 1
         return '%04d' % new_trp_id
 
     def save(self, *args, **kwargs):
         if not self.trp_id:
             self.trp_id = self._get_trp_id()
-        super(Biography, self).save(*args, **kwargs)
+        else:
+            self.trp_id = self.format_trp_id(self.trp_id)        
+        self.roles = self.roles.replace(",",";"); 
+        try:
+            super(Biography, self).save(*args, **kwargs)
+        except IntegrityError as e:
+            print e.message
```

Comment:
Sounds fine to me to just automatically generate a new TRP ID for the user - if we do that, though, I would either remove the option from the user (hide the field on the form, or make it un-editable), or add a note to the user that the TRP ID could/will be changed from what they enter.

I would check with Elli about this, though - there may be some cases where she or others need to change the TRP ID. If some people do need to edit or change the TRP ID, maybe the best option is a note to the user saying it might be changed if it's not unique.